### DISCUSSION TITLE
Greenfield API: Add Webhook for Payouts

### DISCUSSION THREAD

**Discussion by eza** [btcpayserver/btcpayserver#3906] [Technical Support]
In order to improve user experience it would be great to have a webhook for Payouts to let you know when the payout has been processed (executed).

**Comment by bolatovumar** [comment]
Currently all webhook types of invoice states (such as settled, expired, etc...). Adding the "payout processed" (btw, @eza, does "processed" here mean that admin has approved a payout?) would break this convention.

@Kukks do you think it would even make sense to add a webhook for this use case? If it does, then it would probably also make sense to add a webhook type for all types of payout state transitions.

**Comment by eza** [comment]
@bolatovumar processed means that BtcPayServer has executed it and sent the bitcoins to the recipient.

For making sense it makes complete sense. I would like to show the status of the payout in my app to my end user, also I could build notifications to notify users when the I receive the webhook. I personally care only of when the payout has been executed/dispatched, but it would definitely make sense to add all the other types.

**Comment by AustinWildgrube** [comment]
I second this! I have a database table that includes the current pull payment status so users can see it.  On my website, I want to allow the user to start a payout (PullPaymentCreated & PayoutCreated) making my status "Created". Receiving PayoutProcessing changes my status to "Pending," and when PayoutSettled occurs, it changes my status to "Complete". This is kind of what I am thinking in terms of webhooks. This follows the invoice convention and naming.

PullPaymentCreated (Active / Future)
PullPaymentExpired
PullPaymentArchived

PayoutCreated (Awaiting Approval  / Awaiting Payment)
PayoutProcessing (In Progress)
PayoutSettled
PayoutCancelled

**Reply by VIPO777** [reply]
Yes, that would be a convenient solution. I, also need to check the status of the payment, I implemented this by checking the api every time, but this is inconvenient, slow and not very perfomance for me.

**Reply by AustinWildgrube** [reply]
This is what I am going to have to do if it is not implemented. I am trying to stray away from a solution like this, though.

Edit: how exactly did you implement this? Do you just have a loop checking every X seconds/minutes?

**Reply by VIPO777** [reply]
Can say so. I have a task in Redis that will call a method every x minutes to request payment status information in order to, synchronize them with my application's database. I tried to optimize this method, but the implementation of the webhook would have unleashed my hands from this solution.
